### PR TITLE
task for copy claim to header

### PR DESCRIPTION
**Pull request by aryan16**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+title: Copy Jwt Claim to HTTP Header
+description: Shows how users can copy their jwt claims to http headers.
+weight: 30
+keywords: [security,authentication,jwt,claim]
+aliases:
+    - /docs/tasks/security/istio-auth.html
+    - /docs/tasks/security/authn-policy/
+owner: istio/wg-security-maintainers
+test: yes
+status: Experimental
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:11]: Is this feature really experimental? If so, you should also add `{{< boilerplate experimental >}}` at the top of the document text (line 13) as well.

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+---
+This task shows you how to copy valid JWT claims to http headers after JWT authentication is successfully completed via Istio request authentication policy.
+
+Note: Claims of type string, boolean, integer are supported. Array type claims are not supported as of now.
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:15]:
```suggestion
{{< warning >}}
Only claims of type string, boolean, and integer are supported. Array type claims are not supported as of now.
{{< /warning >}}
```

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Before you begin
+
+Before you begin this task, do the following:
+
+* Complete the [Istio end user authentication task](/docs/tasks/security/authentication/authn-policy/#end-user-authentication).
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:21]: This doesn't seem like a necessary step in order to complete this task. Do you really mean something like this:
```suggestion
* Familiarize yourself with [Istio end user authentication](/docs/tasks/security/authentication/authn-policy/#end-user-authentication) support.
```

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+* Install Istio using [Istio installation guide](/docs/setup/install/istioctl/).
+
+* Deploy two workloads: `httpbin` and `sleep`. Deploy these in one namespace,
+for example `foo`. Both workloads run with an Envoy proxy in front of each.
+Deploy the example namespace and workloads using these commands:
+
+    {{< text bash >}}
+    $ kubectl create ns foo
+    $ kubectl apply -f <(istioctl kube-inject -f @samples/httpbin/httpbin.yaml@) -n foo
+    $ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@) -n foo
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:32]: Using `istioctl kube-inject` seems kind of ugly. Why not use auto-injection:
```suggestion
    $ kubectl create ns foo
    $ kubectl label namespace foo istio-injection=enabled
    $ kubectl apply -f @samples/httpbin/httpbin.yaml@ -n foo
    $ kubectl apply -f @samples/sleep/sleep.yaml@ -n foo
```

File: content/en/docs/tasks/security/authentication/claim-to-header/test.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+_verify_same  snip_before_you_begin_2 "$snip_before_you_begin_2_out"
+ 
+snip_allow_requests_with_valid_jwt_and_listtyped_claims_1
+_wait_for_istio requestauthentication foo jwt-example
+ 
+_verify_same snip_allow_requests_with_valid_jwt_and_listtyped_claims_2 "$snip_allow_requests_with_valid_jwt_and_listtyped_claims_2_out"
+ 
+_verify_same snip_allow_requests_with_valid_jwt_and_listtyped_claims_3 "$snip_allow_requests_with_valid_jwt_and_listtyped_claims_3_out"
+ 
+# The previous step stored the JWT in TOKEN, and it's needed in the next step.
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/test.sh:49]: Not sure exactly what this comment is saying. Is this the problem?

```suggestion
# The previous snip stored the JWT in a local (not exported) TOKEN variable, so we need to set it here for the next step.
```

Comment (aryan16) [reply content/en/docs/tasks/security/authentication/claim-to-header/test.sh]: I forgot to remove this comment, it is not required.

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
+---
+title: Copy Jwt Claim to HTTP Header
```
Comment (ericvn) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:2]: We seem to use JWT elsewhere. Should this be uppercased as well as in the description (and maybe elsewhere)?

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+{{< warning >}}
+If you don’t see the expected output, retry after a few seconds.
+Caching and propagation can cause a delay.
+{{< /warning >}}
+
+## Allow requests with valid JWT and list-typed claims
+
+1. The following command creates the `jwt-example` request authentication policy
+for the `httpbin` workload in the `foo` namespace. This policy for `httpbin` workload
+accepts a JWT issued by `testing@secure.istio.io` and copy the value of claim `foo` to a http header `X-Jwt-Claim-Foo`:
```
Comment (ericvn) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:56]: copies?

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    - /docs/tasks/security/istio-auth.html
+    - /docs/tasks/security/authn-policy/
+owner: istio/wg-security-maintainers
+test: yes
+status: Experimental
+---
+This task shows you how to copy valid JWT claims to http headers after JWT authentication is successfully completed via Istio request authentication policy.
+
+{{< warning >}}
+Only claims of type string, boolean, and integer are supported. Array type claims are not supported as of now.
```
Comment (ericvn) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:16]: maybe change as of now to at this time?

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+## Before you begin
+
+Before you begin this task, do the following:
+
+* Familiarize yourself with [Istio end user authentication](/docs/tasks/security/authentication/authn-policy/#end-user-authentication) support.
+
+* Install Istio using [Istio installation guide](/docs/setup/install/istioctl/).
+
+* Deploy two workloads: `httpbin` and `sleep`. Deploy these in one namespace,
+for example `foo`. Both workloads run with an Envoy proxy in front of each.
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:30]:
```suggestion
* Deploy `httpbin` and `sleep` workloads in namespace `foo` with sidecar injection enabled.
```

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+{{< warning >}}
+If you don’t see the expected output, retry after a few seconds.
+Caching and propagation can cause a delay.
+{{< /warning >}}
+
+## Allow requests with valid JWT and list-typed claims
+
+1. The following command creates the `jwt-example` request authentication policy
+for the `httpbin` workload in the `foo` namespace. This policy for `httpbin` workload
+accepts a JWT issued by `testing@secure.istio.io` and copies the value of claim `foo` to a http header `X-Jwt-Claim-Foo`:
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:56]:
```suggestion
for the `httpbin` workload in the `foo` namespace. This policy
accepts a JWT issued by `testing@secure.istio.io` and copies the value of claim `foo` to an http header `X-Jwt-Claim-Foo`:
```

File: content/en/docs/tasks/security/authentication/claim-to-header/index.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+1. Verify that a request with an invalid JWT is denied:
+
+    {{< text bash >}}
+    $ kubectl exec "$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})" -c sleep -n foo -- curl "http://httpbin.foo:8000/headers" -sS -o /dev/null -H "Authorization: Bearer invalidToken" -w "%{http_code}\n"
+    401
+    {{< /text >}}
+
+1. Get the JWT that sets the `iss` and `sub` keys to the same value, `testing@secure.istio.io`.
+This causes Istio to generate the attribute `requestPrincipal` with the value `testing@secure.istio.io/testing@secure.istio.io`:
```
Comment (frankbu) [inline content/en/docs/tasks/security/authentication/claim-to-header/index.md:86]: This is confusing. Why do we care what value Istio generates for the attribute `requestPrincipal`? All the example seems to need is the JWT with issuer `testing@secure.istio.io` and with a `foo` attribute. I would expect that is what this sentence should say this command is doing. Am I missing something?

**Review by frankbu — APPROVED** [review]

**Pull request merged**